tests/stats-calculator.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace Wporg\Tests;
+
+use GP_UnitTestCase;
+use Wporg\TranslationEvents\Stats_Calculator;
+use Wporg\TranslationEvents\Tests\Event_Factory;
+use Wporg\TranslationEvents\Tests\Stats_Factory;
+
+class Stats_Calculator_Test extends GP_UnitTestCase {
+	private Event_Factory $event_factory;
+	private Stats_Factory $stats_factory;
+	private Stats_Calculator $calculator;
+
+	public function setUp(): void {
+		parent::setUp();
+		$this->event_factory = new Event_Factory();
+		$this->stats_factory = new Stats_Factory();
+		$this->calculator    = new Stats_Calculator();
+	}
+
+	public function test_tells_that_event_has_no_stats() {
+		$this->set_normal_user_as_current();
+		$user_id = wp_get_current_user()->ID;
+
+		$event_id = $this->event_factory->create_active( array( $user_id ) );
+		$event    = get_post( $event_id );
+
+		$this->assertFalse( $this->calculator->event_has_stats( $event ) );
+	}
+
+	public function test_tells_that_event_has_stats() {
+		$this->set_normal_user_as_current();
+		$user_id = wp_get_current_user()->ID;
+
+		$event_id = $this->event_factory->create_active( array( $user_id ) );
+		$event    = get_post( $event_id );
+
+		$this->stats_factory->create( $event_id, $user_id, 1, 'create' );
+
+		$this->assertTrue( $this->calculator->event_has_stats( $event ) );
+	}
+
+	public function test_calculates_stats_for_event() {
+		$this->set_normal_user_as_current();
+		$user1_id = 42;
+		$user2_id = 43;
+		$user3_id = 44;
+
+		$event1_id = $this->event_factory->create_active( array( $user1_id ) );
+		$event2_id = $this->event_factory->create_active( array( $user1_id ) );
+
+		// For event1, aa locale, multiple users.
+		$this->stats_factory->create( $event1_id, $user1_id, 11, 'create' );
+		$this->stats_factory->create( $event1_id, $user2_id, 11, 'create' );
```

Comment:
How could a user create two translations with the same id? I understand it's possible from a data perspective but it should not happen in reality.

Reply:
Makes sense, fixed.